cartography/intel/gcp/apigateway.py

```diff
@@ -96,9 +100,67 @@ def get_apis(apigateway: Resource, project_id: str) -> List[Dict]:
         else:
             raise
 
+@timeit
+def get_api_policy_users(apigateway: Resource, apis: List[Resource],project_id: str) -> List[Dict]:
+    """
+        Returns a list of users attached to IAM policy of an API within the given project.
+
+        :type apigateway: The GCP APIGateway resource object
+        :param apigateway: The APIGateway resource object created by googleapiclient.discovery.build()
+
+        :type api: List
+        :param api: The list of APIs
+
+        :type project_id: str
+        :param project_id: Current Google Project Id
+
+        : type locations: List
+        : param Location: List of locations
+
+        :rtype: list
+        :return: List of api iam policy users
+    """
+    users = []
+    try:
+        for api in apis:
+            api['iam_policy'] = apigateway.projects().locations().apis().getIamPolicy(\
+                resource=api['id']).execute()
+            bindings = api.get('iam_policy',{}).get('bindings',[])
+            for binding in bindings:
+                for member in binding['members']:
+                    if member.startswith('user:'):
```

Comment:
we should do this for all types of members (users, roles, service accounts, etc.), not just users. Some services do not attach user level permissions. For Ex: GCP Functions always get attached to a service account. Most likely, these will never get associated with user or role.